### PR TITLE
silx.gui.colors: Added typing

### PR DESCRIPTION
**Pull request by t20100**
This PR adds typing to `silx.gui.colors` and defines input and output types for `colors.rgba`.
Since this function is used everywhere for colors, it should help to type colors in the plot API.

### CODE REVIEW

File: src/silx/gui/colors.py [hunk 1]
```diff
@@ -238,21 +249,46 @@ def _getColormap(name):
     return _colormap.get_colormap_lut(name)
 
 
+class _Colormappable:
```
Comment (t20100) [inline src/silx/gui/colors.py:252]: Base class for objects that interacts with `Colormap` i.e, `ColormapMixIn` plot items: This avoids lazy imports and ease typing.

**Comment by payno** [comment]
Looks good. Can we merge it ?

**Comment by t20100** [comment]
Yes, if you approuve it ;)

**Comment by payno** [comment]
I am all for typing :)

**Pull request merged**